intranet/apps/auth/tests.py

```diff
@@ -27,12 +30,70 @@ def test_login_page(self):
+        block.save()
+        return block
+
+    def add_activity(self, **args):
+        activity = EighthActivity(**args)
+        activity.save()
+        return activity
+
+    def schedule_activity(self, block, activity):
+        schact = EighthScheduledActivity(block=block, activity=activity)
+        schact.save()
+        return schact
+
+    def login_student(self):
+        user = User.objects.get_or_create(username="awilliam")[0]
+        user.user_type = "student"
+        user.first_login = timezone.now()
+        user.seen_welcome = True
+        user.save()
         with self.settings(MASTER_PASSWORD='pbkdf2_sha256$24000$qp64pooaIEAc$j5wiTlyYzcMu08dVaMRus8Kyfvn5ZfaJ/Rn+Z/fH2Bw='):
-            response = self.client.post(reverse("index"), data={
+            return self.client.post(reverse("login"), data={
                 "username": "awilliam",
                 "password": "dankmemes"
             })
 
-        self.assertEqual(response.status_code, 200)
+    def does_login_redirect_to(self, url):
+        response = self.login_student()
+        return response.status_code == 302 and response["Location"] == url
+
+    def test_authentication(self):
+        self.assertTrue(self.does_login_redirect_to(reverse("index")))
+
+    def test_eighth_login_redirect(self):
+        now = timezone.localtime(timezone.now())
+
+        self.login_student()
+        user = User.objects.get(username="awilliam")
+
+        # Don't let blocks created in other tests contaminate these results
+        for block in EighthBlock.objects.all():
+            block.delete()
+
+        deltas = {minutes: datetime.timedelta(minutes=minutes) for minutes in (-5, 5, 15)}
+
+        activity = self.add_activity(name="Test Activity 1")
+
+        with self.settings(ENABLE_PRE_EIGHTH_REDIRECT=True):
+            block_15 = self.add_block(date=now.date(), signup_time=(now + deltas[15]).time(), block_letter="A")
+            self.assertTrue(self.does_login_redirect_to(reverse("index")))
+            EighthSignup(user=user, scheduled_activity=self.schedule_activity(block_15, activity)).save()
+            self.assertTrue(self.does_login_redirect_to(reverse("index")))
+
+            block_5 = self.add_block(date=now.date(), signup_time=(now + deltas[5]).time(), block_letter="B")
+            self.assertTrue(self.does_login_redirect_to(reverse("eighth_signup")))
+            EighthSignup(user=user, scheduled_activity=self.schedule_activity(block_5, activity)).save()
+            self.assertTrue(self.does_login_redirect_to(reverse("index")))
+
+            block_n5 = self.add_block(date=now.date(), signup_time=(now + deltas[-5]).time(), block_letter="C")
+            self.assertTrue(self.does_login_redirect_to(reverse("index")))
+            EighthSignup(user=user, scheduled_activity=self.schedule_activity(block_n5, activity)).save()
+            self.assertTrue(self.does_login_redirect_to(reverse("index")))
+
+            block_15.delete()
```

Comment:
If you want to delete all EighthBlock objects in one swoop, you can use `EighthBlock.objects.all().delete()`